auth/types.go

```diff
@@ -183,3 +185,58 @@ type CertificateQuery struct {
 	// Subject identity key (who the certificate is about)
 	Subject string
 }
+
+func (m *AuthMessage) MarshalJSON() ([]byte, error) {
```

Comment:
It would be good to have tests for Marchaling and Unmarshaling if it's compatible.

Reply:
Sure, will add today